presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -110,6 +108,18 @@ public void matches(PlanMatchPattern pattern)
                     actual.getOutputSymbols()));
         }
 
-        assertPlan(session, metadata, lookup, new Plan(actual, types), pattern);
+        executeInTransaction(queryRunner, session -> {
+            assertPlan(session, queryRunner.getMetadata(), lookup, new Plan(actual, types), pattern);
+            return null;
+        });
+    }
+
+    private <T> T executeInTransaction(LocalQueryRunner queryRunner, Function<Session, T> transactionFunction)
+    {
+        return queryRunner.inTransaction(session ->
+        {
+            session.getCatalog().ifPresent(catalog -> queryRunner.getMetadata().getCatalogHandle(session, catalog));
```

Comment:
yeah.  It's side effect is registering the catalog for the transaction, which is pretty weird, but that's how everything else registers the catalog.   (see TransactionManager.getConnectorId())